src/starknet.rs

```diff
@@ -251,10 +251,10 @@ pub trait StarknetSyscallHandler {
 
     fn sha256_process_block(
         &mut self,
-        prev_state: &[u32; 8],
-        current_block: &[u32; 16],
+        state: &mut [u32; 8],
+        block: &[u32; 16],
         remaining_gas: &mut u128,
-    ) -> SyscallResult<[u32; 8]>;
+    ) -> SyscallResult<()>;
```

Comment:
q: why was this modified? isn't this going to be a problem for upstream blockifier? (I had to adapt my types so i'm wondering)

Reply:
There are various reasons:
  - The technical reason is that internally the state is represented as a box. We receive a box and return a box so we're just reusing the allocation.
  - All libraries I've seen mutate the state instead of returning a new one.
  - We can simplify the MLIR code by avoiding (stack) `alloca`s and `memcpy`s.

Before this we `malloc`'d a new state, passed the old state's pointer as the current state and the new pointer as the return value's location, which meant that the previous state's allocation got leaked. Now it doesn't get leaked and we do not need extra allocations.

Our stub syscall handler has an example implementation with the changes to adapt it [here](https://github.com/lambdaclass/cairo_native/pull/833/files/b73e3fa8927eb284d362594aeaff900e28636a8d#diff-b9b3ae981ddf1b61c79d36505c39e96f0ad175b042d384f1e1d9d637a1034991R823-R837).